huaweicloud/services/vod/resource_huaweicloud_vod_transcoding_template_group.go

```diff
@@ -0,0 +1,576 @@
+											validation.IntBetween(128, 2160),
+										),
+									},
+									"bitrate": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										ValidateFunc: validation.Any(
+											validation.IntInSlice([]int{0}),
+											validation.IntBetween(700, 3000),
+										),
+									},
+									"frame_rate": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										ValidateFunc: validation.IntBetween(1, 75),
+										Default:      1,
+									},
+								},
+							},
+						},
+						"audio": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"sample_rate": {
+										Type:         schema.TypeInt,
+										Required:     true,
+										ValidateFunc: validation.IntBetween(1, 6),
+									},
+									"channels": {
+										Type:         schema.TypeInt,
+										Required:     true,
+										ValidateFunc: validation.IntBetween(1, 2),
+									},
+									"bitrate": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										ValidateFunc: validation.Any(
+											validation.IntInSlice([]int{0}),
+											validation.IntBetween(8, 1000),
+										),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildCommonOpts(rawCommon []interface{}) *vod.Common {
+	if len(rawCommon) != 1 {
+		return nil
+	}
+
+	common := rawCommon[0].(map[string]interface{})
+	commonOpts := vod.Common{
+		HlsInterval: utils.Int32(int32(common["hls_segment_duration"].(int))),
+	}
+
+	if common["low_bitrate_hd"].(bool) {
+		commonOpts.Pvc = vod.GetCommonPvcEnum().E_1
+	} else {
+		commonOpts.Pvc = vod.GetCommonPvcEnum().E_0
+	}
+
+	var videoCodec vod.CommonVideoCodec
+	if common["video_codec"].(string) == "H265" {
+		videoCodec = vod.GetCommonVideoCodecEnum().H265
+	} else {
+		videoCodec = vod.GetCommonVideoCodecEnum().H264
+	}
+	commonOpts.VideoCodec = &videoCodec
+
+	var audioCodec vod.CommonAudioCodec
+	switch common["audio_codec"].(string) {
+	case "HEAAC1":
+		audioCodec = vod.GetCommonAudioCodecEnum().HEAAC1
+	case "HEAAC2":
+		audioCodec = vod.GetCommonAudioCodecEnum().HEAAC2
+	case "MP3":
+		audioCodec = vod.GetCommonAudioCodecEnum().MP3
+	default:
+		audioCodec = vod.GetCommonAudioCodecEnum().AAC
+	}
+	commonOpts.AudioCodec = &audioCodec
+
+	return &commonOpts
+}
+
+func buildAudioOpts(rawAudio []interface{}) *vod.AudioTemplateInfo {
+	if len(rawAudio) != 1 {
+		return nil
+	}
+
+	audio := rawAudio[0].(map[string]interface{})
+	audioOpts := vod.AudioTemplateInfo{
+		SampleRate: int32(audio["sample_rate"].(int)),
+		Channels:   int32(audio["channels"].(int)),
+		Bitrate:    utils.Int32(int32(audio["bitrate"].(int))),
+	}
+
+	return &audioOpts
+}
+
+func buildVideoOpts(rawVideo []interface{}) *vod.VideoTemplateInfo {
+	if len(rawVideo) == 0 {
```

Comment:
suggest to keep this logic consistent with other and easier to read.

Reply:
fixed